### PR TITLE
DB-8846 changes in CDH-installation.md: Java Configuration Options for HBase Master

### PR DESCRIPTION
**Pull request by martinrupp**
(first version of this only added to olapServer.classPath)

### CODE REVIEW

**Comment by martinrupp** [comment]
(this has been checked on srv131 to work)

**Comment by martinrupp** [comment]
> There is no /opt/cloudera/parcels/SPARK2 in CDH6.3.0 any more (SPARK2 is not a separate parcel but part of CDH). All those paths should be fixed.

OK! We have this line here
-Dsplice.spark.yarn.jars=/opt/cloudera/parcels/SPARK2/lib/spark2/jars/*

should this be removed completely?

**Comment by OlegMazurov** [comment]
There are two places where SPARK2 is used. You can check with srv131 configuration to see what we actually use with CDH 6.x.

**Comment by martinrupp** [comment]
> There are two places where SPARK2 is used. You can check with srv131 configuration to see what we actually use with CDH 6.x.

ok i checked on srv131. changes are:

/opt/cloudera/parcels/SPARK2/lib/spark2/jars/
->
/opt/cloudera/parcels/CDH/lib/spark/jars/

in two places. will amend commit

**Comment by martinrupp** [comment]
see conversation on slack about `-XX:MaxPermSize=512`, `-Dsplice.spark.master=yarn-client` to `-Dsplice.spark.master=yarn` and `-Dspark.compaction.reserved.slots=4`

File: platforms/cdh6.3.0/docs/CDH-installation.md [hunk 1]
```diff
@@ -550,13 +550,13 @@ com.splicemachine.hbase.BackupEndpointObserver</code>
...
...th=/opt/cloudera/parcels/CDH/lib/hbase/conf:/opt/cloudera/parcels/CDH/jars/htrace-core-3.1.0-incubating.jar -Dsplice.spark.executor.extraLibraryPath=/opt/cloudera/parcels/CDH/lib/hbase/conf:/opt/cloudera/parcels/CDH/lib/hadoop/lib/native -Dsplice.olapServer.classpath=/opt/cloudera/parcels/CDH/lib/hive/lib/* -Dsplice.spark.executor.extraClassPath=/opt/cloudera/parcels/CDH/lib/hbase/conf:/opt/cloudera/parcels/SPLICEMACHINE/lib/*:/opt/cloudera/parcels/CDH/lib/spark/jars/*:/opt/cloudera/parcels/CDH/lib/hbase/lib/*:/opt/cloudera/parcels/CDH/lib/hive/lib/* -Dsplice.spark.executor.extraLibraryPath=/opt/cloudera/parcels/CDH/lib/hadoop/lib/native -Dsplice.spark.ui.retainedJobs=100 -Dsplice.spark.ui.retainedStages=100 -Dsplice.spark.worker.ui.retainedExecutors=100 -Dsplice.spark.worker.ui.retainedDrivers=100 -Dsplice.spark.streaming.ui.retainedBatches=100 -Dsplice.spark.executor.cores=10 -Dsplice.spark.executor.memory=24g -Dspark.compaction.reserved.slots=4  -Dsplice.spark.local.dir=/tmp -Dsplice.spark.yarn.jars=/opt/cloudera/parcels/CDH/lib/spark/jars/* -Dsplice.spark.sql.shuffle.partitions=900 -Dsplice.spark.reducer.maxSizeInFlight=512m -Dsplice.spark.maxRemoteBlockSizeFetchToMem=105000000
```
Comment (dgomezferro) [inline platforms/cdh6.3.0/docs/CDH-installation.md:553]: > -Dsplice.spark.master=yarn

This looks good but it's missing the deploy-mode, I would explicitly list it even though the default is client: `-Dsplice.spark.submit.deployMode=client`

Comment (jyuanca) [reply platforms/cdh6.3.0/docs/CDH-installation.md:553]: Please remove -Dsplice.spark.shuffle.compress=false

Comment (jyuanca) [reply platforms/cdh6.3.0/docs/CDH-installation.md:553]: Why -Dsplice.olapServer.classpath=/opt/cloudera/parcels/CDH/lib/hive/lib/*?

Comment (martinrupp) [reply platforms/cdh6.3.0/docs/CDH-installation.md]: > > -Dsplice.spark.master=yarn
> 
> This looks good but it's missing the deploy-mode, I would explicitly list it even though the default is client: `-Dsplice.spark.submit.deployMode=client`

done

> Please remove -Dsplice.spark.shuffle.compress=false
done

> Why -Dsplice.olapServer.classpath=/opt/cloudera/parcels/CDH/lib/hive/lib/*

i don't know about this one, this was already done like that before, and is also used e.g. on srv131

**Comment by martinrupp** [comment]
New change does the following:

- classpath has to be added both for olapServer AND spark.executor.extraClassPath
- added `-Dsplice.spark.submit.deployMode=client`
- removed `-Dsplice.spark.shuffle.compress=false`
- `-Dsplice.spark.master=yarn-client` -> `-Dsplice.spark.master=yarn`
- changed
`/opt/cloudera/parcels/SPARK2/lib/spark2/jars/`
->
`/opt/cloudera/parcels/CDH/lib/spark/jars/`

Please see comparision srv131 vs CDH-Installation here (Sheet "v3")
https://docs.google.com/spreadsheets/d/1jqa_Qku7qyDGBHvOaPZl5mYFZMEnGeH-lGIJHr2hnU0/edit?usp=sharing#gid=1360568476

Changes are currently (here=the CDH-Installation.md config):

- `-Dsplice.spark.driver.maxResultSize`:  0 on srv131, here 2g 
- `-Dsplice.spark.dynamicAllocation.cachedExecutorIdleTimeout` and `-Dsplice.spark.dynamicAllocation.executorIdleTimeout`: 1200 on srv131, 120 here.
- `-Dsplice.spark.dynamicAllocation.minExecutors`: 3 on srv131, here 4
- `-Dsplice.spark.executor.memory`: 32500m on srv131, here 24g
- `-Dsplice.spark.maxRemoteBlockSizeFetchToMem`: 83886080 on srv131, here 105000000
- `-Dsplice.spark.reducer.maxSizeInFlight`: 400m on srv131, here 512m
- `-Dsplice.spark.ui.retainedJobs`: 150 on srv131, 100 here
- `-Dsplice.spark.ui.retainedStages`: 250 on srv131, 100 here
- `-Dsplice.spark.yarn.am.memory`: 2g on srv131, 1g here

only on srv131:
- `-Dsplice.spark.executor.memoryOverhead=4096`
- `-Dsplice.spark.yarn.containerLauncherMaxThreads=12`

only here:
- `-Dsplice.spark.local.dir=/tmp`
- `-Dsplice.spark.sql.shuffle.partitions=900`
- `-Dsplice.spark.yarn.executor.memoryOverhead=8192`

Sheet "old vs. new" shows the changes we do in this change.

**Comment by martinrupp** [comment]
@OlegMazurov said `-Dspark.compaction.reserved.slots=4` is propably wrong, should i remove it?

**Comment by martinrupp** [comment]
> @OlegMazurov said `-Dspark.compaction.reserved.slots=4` is propably wrong, should i remove it?

@dgomezferro said this should be removed.

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by yxia92** [comment]
@martinrupp Is this PR specific to master? I don't see PRs for branch-3.0 and branch-2.8. Thanks!

**Comment by martinrupp** [comment]
> +1
> 
> Please apply these changes to release 3.0 as well.
done.

- 3.0: https://github.com/splicemachine/spliceengine/pull/3711
- 2.8: https://github.com/splicemachine/spliceengine/pull/3712

note that 2.8 contains only some of the 3.0/master changes because ORC was only a problem on CDH6.3.0. i aligned the other configurations.

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Comment by martinrupp** [comment]
com.splicemachine.hbase.SpliceRegionAdminIT.testMultipleMerge seems to be sporadic (2nd time i've seen this)

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**